joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodReturnTests.scala

```diff
@@ -315,4 +315,76 @@ class MethodReturnTests extends RubyCode2CpgFixture(withDataFlow = true) {
     }
   }
 
+  "implict RETURN node for RubyCallWithBlock" in {
+    val cpg = code("""
+        | def foo &block
+        |  puts block.call
+        | end
+        |
+        | def bar
+        |  foo do
+        |   "hello"
+        |  end
+        | end
+        |""".stripMargin)
+
+    inside(cpg.method.name("bar").l) {
+      case bar :: Nil =>
+        inside(bar.block.astChildren.collectAll[Method].l) {
+          case closureMethod :: Nil =>
+            closureMethod.name shouldBe "<lambda>0"
+            closureMethod.fullName shouldBe "Test0.rb:<global>::program:bar:<lambda>0"
+          case xs => fail(s"Expected closure method, but found ${xs.code.mkString(", ")} instead")
+        }
+
+        inside(bar.methodReturn.toReturn.l) {
+          case barReturn :: Nil =>
+            inside(barReturn.astChildren.l) {
+              case (returnCall: Call) :: Nil =>
+                returnCall.code.replaceAll("\n", "") shouldBe "foo do   \"hello\"  end"
```

Comment:
Could you test the arguments and name here too? We just need to verify that the method ref is given